File: CITS3002 - Networks/Project - Transport Station - Client and Server/C++Project/ConnLog.cpp

#include "globals.h"

/**
 * Create a connection log from a string representation
 *
 * @param strRep
 */
ConnLog::ConnLog(String strRep) {

    // If empty
    if (strEquals(strRep, "Empty")) {

        // Do not process further
        return;
    }

    // Split into connection representations
    vector<String> connsS = split(strRep, sep);

    // For each route stop string
    for (String connStr : connsS) {

        // Extract route stop and add to log
        conns.push_back(Connection(connStr));
    }
}

/**
 * Get string representation
 *
 * @return
 */
String ConnLog::toString() {

    // If log is empty
    if (conns.empty()) {

        // Notify
        return "Empty";
    }

    // Holder
    String rep = "";

    // Add route stops separated
    for (Connection curConn : conns) {

        rep += curConn.toString() + sep;
    }

    // Return representation
    return rep;
}

/**
 * Get the first connection
 *
 * @return
 */
Connection ConnLog::getFirst() {
    return conns.at(0);
}

/**
 * Get the last connection
 *
 * @return
 */
Connection ConnLog::getLast() {
    return conns.at(conns.size() - 1);
}

/**
 * Get separator
 * @return 
 */
String ConnLog::getSep() {
    return sep;
}

// ----
// ------------------------------------
// ----

/**
 * Constructor
 */
ConnLog::ConnLog() {
}

/**
 * Special
 * @param orig
 */
ConnLog::ConnLog(const ConnLog& orig) {
}

/**
 * Destructor
 */
ConnLog::~ConnLog() {
}

File: CITS3002 - Networks/Project - Transport Station - Client and Server/JavaProject/src/statgram/log/ConnLog.java
package statgram.log;

import java.util.ArrayList;
import timetable.Connection;

/**
 * Represents a log of connections
 *
 * @author David
 */
public class ConnLog extends ArrayList<Connection> {

    // Separator
    public static final String sep = ">>";

    /**
     * Create an empty connection log
     */
    public ConnLog() {

    }

    /**
     * Create a connection log from a string representation
     *
     * @param strRep
     */
    public ConnLog(String strRep) {

        // If empty
        if (strRep.equals("Empty")) {

            // Do not process further
            return;
        }

        // Split into connection representations
        String[] connsS = strRep.split(sep);

        // For each connection string
        for (String connStr : connsS) {

            // Extract connection and add to log
            this.add(new Connection(connStr));
        }
    }

    /**
     * Get string representation
     *
     * @return
     */
    @Override
    public String toString() {

        // If log is empty
        if (isEmpty()) {

            // Notify
            return "Empty";
        }

        // Holder
        String rep = "";

        // Add connections separated
        for (Connection curConn : this) {

            rep += curConn.toString() + sep;
        }

        // Return representation
        return rep;
    }

    /**
     * Get the last connection
     *
     * @return
     */
    public Connection getLast() {
        return get(size() - 1);
    }
}
